rust/src/lib.rs

```diff
@@ -3173,6 +3174,69 @@ pub unsafe extern "C" fn zcashlc_create_proposed_transactions(
     unwrap_exc_or_null(res)
 }
 
+/// Metadata about the status of a transaction obtained by inspecting the chain state.
+#[repr(C, u8)]
+pub enum FfiTransactionStatus {
+    /// The requested transaction ID was not recognized by the node.
+    TxidNotRecognized,
+    /// The requested transaction ID corresponds to a transaction that is recognized by the node,
+    /// but is in the mempool or is otherwise not mined in the main chain (but may have been mined
+    /// on a fork that was reorged away).
+    NotInMainChain,
+    /// The requested transaction ID corresponds to a transaction that has been included in the
+    /// block at the provided height.
+    Mined(u32),
+}
```

Comment:
By comparison, in [the Android SDK](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/files#diff-63f3f4a8b98989699289a769aae68d9a8671559608f0119647df4a07194fcab3R1686-R1690) I passed sentinel values across because JNI doesn't have an enum type I could use. This is much nicer.